lib/Artwork.php

```diff
@@ -618,8 +623,19 @@ public function Create(array $uploadedFile): void{
 	 * @throws \Exceptions\ValidationException
 	 */
 	public function Save(): void{
+		$this->_UrlName = null;
+
 		$this->Validate();
 
+		// Can't assign directly to $this->Tags because it's hidden behind a getter
+		$tags = [];
+		foreach($this->Tags as $artworkTag){
+			$tags[] = ArtworkTag::GetOrCreate($artworkTag);
+		}
+		$this->Tags = $tags;
+
+		$this->Artist = Artist::GetOrCreate($this->Artist);
```

Comment:
I believe `GetOrCreate()` is doing the correct thing here, modulo a few edge cases.

1. If the user updates the artist from `John Nobody` to `John Q. Nobody`, then yes we will create a new record and the old one will be a zombie. It would have been better to update the `Artist` record. But if the user updates the artist from `Claude Monet` to `John Q. Nobody`, we don't want to update `Artist` record. At this point in the code, it's hard to tell those two cases apart because we've already lost the old `Artist`. If the user updates the artist from `Claude Monet` to `John Singer Sargent`, `Artist::GetOrCreate()` works as expected and looks up the existing artist.

2. It's not possible to edit just the artist's death year on that form. This is consistent with the advice on the form, "_For existing artists, leave the year of death blank._"

Let me know if you'd like me to try harder here.

Reply:
OK. So what we should do is after updating the artist, delete any artists in the DB who are not attached to an artwork.

Reply:
Yep, good call, done. It was a straightforward implementation, too.